src/api/routes/rosetta/construction.ts

```diff
@@ -604,6 +643,60 @@ export function createRosettaConstructionRouter(db: DataStore, chainId: ChainID)
         };
         transaction = await makeUnsignedContractCall(stackingTx);
         break;
+      }
+      case 'delegate-stacking': {
+        const poxBTCAddress = publicKeyToBitcoinAddress(
+          publicKeys[0].hex_bytes,
+          req.body.network_identifier.network
+        );
+        if (!poxBTCAddress) {
+          res.status(500).json(RosettaErrors[RosettaErrorsTypes.invalidPublicKey]);
+          return;
+        }
+        const { version, hash } = btcAddress.fromBase58Check(poxBTCAddress);
+        const versionBuffer = bufferCV(new BN(version, 10).toBuffer());
+        const hashbytes = bufferCV(hash);
+        const poxAddressCV = tupleCV({
+          hashbytes,
+          version: versionBuffer,
+        });
```

Comment:
I'm not sure this works. Shouldn't the pox btc address be created with the `@stacks/stacking` lib which will convert the `version` byte?
cc @kyranjamie

Reply:
In the Stacks Wallet we don't let users define a BTC address, as this is managed by the pool. Would be a good idea to use the stacking lib if we can.

I'd also recommend these `case` statements match the `functionName` being called:

- `delegate-stx`
- `stack-stx`

Reply:
@rllola the above code for `stack-stx` is also broken. It also needs to convert the btc address to a pox address with the correct version byte. See how it's done in the `@stacks/stacking` lib: https://github.com/blockstack/stacks.js/blob/ce27b48c31b3101b0faf2360fa07655f0ec35fe1/packages/stacking/src/index.ts#L488-L511

Reply:
@zone117x I don't see the difference between what is in the PR and what you linked ? The pox btc address in converted to a pox address.

Reply:
Ah I think I got it. It is a the `decodeBtcAddress` function that should be used ? Instead of getting the btc address version

Reply:
The code here is incorrectly using the `version` byte from returned from `bitcoinjs.fromBase58Check`. The code in the `@stacks/stacking` lib converts the `version` byte to a Stacks address hash mode byte.

Reply:
It should be fixed now.